crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -180,4 +193,106 @@ impl<'a> Compressor<'a> {
 
         None
     }
+
+    fn try_fold_unary_operator<'b>(
+        &mut self,
+        unary_expr: &'b mut UnaryExpression<'a>,
+    ) -> Option<Expression<'a>> {
+        self.fold_expression(&mut unary_expr.argument);
+
+        if let Some(boolean) = get_boolean_value(&unary_expr.argument) {
+            match unary_expr.operator {
```

Comment:
For each of these cases, can you write some comments on what the fold does, e.g.`xxx -> yyy`, so we don't get lost on all the details?